flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -212,7 +217,9 @@ public void testBackwardsEventWithUninitializedChannel() throws Exception {
 		verify(taskEventDispatcher, times(1)).publish(any(ResultPartitionID.class), any(TaskEvent.class));
 
 		// After the update, the pending event should be send to local channel
-		inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(new ResultPartitionID(unknownPartitionId.getPartitionId(), unknownPartitionId.getProducerId()), ResultPartitionLocation.createLocal()));
+
+		ResourceID location = ResourceID.generate();
```

Comment:
It seems we could use the default `ResourceID` in `NettyShuffleDescriptorBuilder`

Reply:
I think the random one provides a better check for tests.